### PR TITLE
Created HermaFx.DataAnnotations.DistinctElementsAttribute

### PR DESCRIPTION
**Pull request by fravalpin**
A new attribute was created :)

### CODE REVIEW

File: HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+		{
+			var model = new Model();
+
+			model.Values = new List<Model._Item>()
+			{
+					new Model._Item() { Property1 = "hello", Property2 = "world" },
+					new Model._Item() { Property1 = "hi", Property2 = "world" },
+			};
+
+			Assert.That(ExtendedValidator.IsValid(model), Is.True);
```
Comment (pruiz) [inline HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs:38]: Please, add some asserts again Validation ensuring thrown exception and/or ValidationResult points to correct named property / element duplicated. Also ensure:

1. Correct (default) message is fired along exception/result.
2. Correct (overriden) message is fired along exception/result when overriding message (thru attribute's ErrorMessage & ErrorMessageResource properties)

File: HermaFx.DataAnnotations/DistinctElementsAttribute.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+namespace HermaFx.DataAnnotations
+{
+	[AttributeUsage(AttributeTargets.Property)]
+	public class DistinctElementsAttribute : ValidationAttribute
+	{
+		public string UniqueProperty { get; }
+		public Type ElementType { get; }
+		public DistinctElementsAttribute(string property)
+		{
+			ErrorMessage = "The property {0} must be unique in the collection.";
```
Comment (pruiz) [inline HermaFx.DataAnnotations/DistinctElementsAttribute.cs:16]: Does the actual property collection name get shown somewhere in the thrown exception?

**Review by pruiz — CHANGES_REQUESTED** [review]
It would be good to have a few more test coverage..

File: HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs (hunk 1 above)
Comment (fravalpin) [reply HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs:40]: Done. Is it Ok?

File: HermaFx.DataAnnotations/DistinctElementsAttribute.cs (hunk 2 above)
Comment (fravalpin) [reply HermaFx.DataAnnotations/DistinctElementsAttribute.cs]: You're right. I've added it.

File: HermaFx.DataAnnotations/DistinctElementsAttribute.cs [hunk 3]
```diff
@@ -66,7 +66,7 @@ protected override ValidationResult IsValid(object value, ValidationContext cont
 				return ValidationResult.Success;
 			}
 
-			return new ValidationResult(FormatErrorMessage(UniqueProperty));
+			return new ValidationResult(string.Format(ErrorMessage, UniqueProperty, context.MemberName));
```
Comment (pruiz) [inline HermaFx.DataAnnotations/DistinctElementsAttribute.cs:69]: I think you need to actually use FormatErrorMessage, as it takes care of using the fixed message passed as ErrorMessage, but also supports ErrorMessageResourceName (or whtaever it is named), etc.

Regarding passing additional format parameters to format error message.. I dont know / remember now the appropiate way, but we may want to take a look a source code or examples of other attribures with similiar issues.

File: HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs (hunk 1 above)
Comment (pruiz) [reply HermaFx.DataAnnotations.Tests/DistinctElementsAttributeTest.cs:40]: Actually I think you current changes test ErrorMessage, but break ErrorMessageResourceName.. :?

**Review by pruiz — CHANGES_REQUESTED** [review]

File: HermaFx.DataAnnotations/DistinctElementsAttribute.cs (hunk 3 above)
Comment (fravalpin) [reply HermaFx.DataAnnotations/DistinctElementsAttribute.cs:69]: could it be by overriding the FormatErrorMessage method?
```csharp
public override string FormatErrorMessage(string name)
{
    return string.Format(ErrorMessageString, UniqueProperty, name);
}
```

Comment (pruiz) [reply HermaFx.DataAnnotations/DistinctElementsAttribute.cs:69]: I guess so..

Comment (fravalpin) [reply HermaFx.DataAnnotations/DistinctElementsAttribute.cs]: its already changed

**Pull request merged**